File: src/storage.py
import json
from pathlib import Path

from .config import MAX_CHUNK_CHARS, OVERLAP_CHARS, TARGET_CHUNK_CHARS


def save_markdown(markdown: str, output_path: Path) -> None:
    """Save converted Markdown to disk."""
    output_path.parent.mkdir(parents=True, exist_ok=True)
    output_path.write_text(markdown, encoding="utf-8")
    print(f"Markdown saved: {output_path}")


def save_chunks(chunks: list[dict], output_path: Path) -> None:
    """Save chunks and their configuration as formatted JSON."""
    output_path.parent.mkdir(parents=True, exist_ok=True)
    data = {
        "total_chunks": len(chunks),
        "chunking_config": {
            "target_chunk_chars": TARGET_CHUNK_CHARS,
            "max_chunk_chars": MAX_CHUNK_CHARS,
            "overlap_chars": OVERLAP_CHARS,
        },
        "chunks": chunks,
    }
    output_path.write_text(
        json.dumps(data, indent=2, ensure_ascii=False),
        encoding="utf-8",
    )
    print(f"Chunks saved: {output_path}")

File: web/lib/src/storage.ts
import { mkdir, writeFile } from "fs/promises";
import { dirname } from "path";
import type { Chunk } from "./types";
import {
  MAX_CHUNK_CHARS,
  OVERLAP_CHARS,
  TARGET_CHUNK_CHARS,
} from "./config";



export async function saveMarkdown(
  markdown: string,
  outputPath: string
): Promise<void> {
  await mkdir(dirname(outputPath), {
    recursive: true,
  });

  await writeFile(outputPath, markdown, {
    encoding: "utf-8",
  });

  console.log(`Markdown saved: ${outputPath}`);
}

export async function saveChunks(
  chunks: Chunk[],
  outputPath: string
): Promise<void> {
  await mkdir(dirname(outputPath), {
    recursive: true,
  });

  const data = {
    total_chunks: chunks.length,

    chunking_config: {
      target_chunk_chars: TARGET_CHUNK_CHARS,
      max_chunk_chars: MAX_CHUNK_CHARS,
      overlap_chars: OVERLAP_CHARS,
    },

    chunks,
  };

  await writeFile(
    outputPath,
    JSON.stringify(data, null, 2),
    {
      encoding: "utf-8",
    }
  );

  console.log(`Chunks saved: ${outputPath}`);
}

export async function saveJson(
  data: unknown,
  outputPath: string,
  label: string
): Promise<void> {
  await mkdir(dirname(outputPath), {
    recursive: true,
  });

  await writeFile(
    outputPath,
    JSON.stringify(data, null, 2),
    {
      encoding: "utf-8",
    }
  );

  console.log(`${label} saved: ${outputPath}`);
}